scaladoc-js/src/versions-dropdown/DropdownHandler.scala

```diff
@@ -0,0 +1,94 @@
+package dotty.tools.scaladoc
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.{Success,Failure}
+
+import org.scalajs.dom._
+import org.scalajs.dom.ext._
+import scala.scalajs.js.annotation.JSExportTopLevel
+import org.scalajs.dom.ext.Ajax
+import scala.scalajs.js
+import scala.scalajs.js.JSON
+
+trait Versions extends js.Object:
+  def versions: js.Dictionary[String]
+
+class DropdownHandler:
+
+  val KEY = "versions-json"
+  val UNDEFINED_VERSIONS = "undefined_versions"
+
+  private def addVersionsList(json: String) =
+    val ver = JSON.parse(json).asInstanceOf[Versions]
+    val ddc = document.getElementById("dropdown-content")
+    for (k, v) <- ver.versions do
+      var child = document.createElement("a").asInstanceOf[html.Anchor]
+      child.href = v
+      child.text = k
+      ddc.appendChild(child)
+    val arrow = document.createElement("span").asInstanceOf[html.Span]
+    arrow.classList.add("ar")
+    document.getElementById("dropdown-button").appendChild(arrow)
+
+  private def disableButton() =
+    val btn = document.getElementById("dropdown-button").asInstanceOf[html.Button]
+    btn.disabled = true
+    btn.classList.remove("dropdownbtnactive")
+
+  private def getURLContent(url: String): Future[String] = Ajax.get(url).map(_.responseText)
+
+  window.sessionStorage.getItem(KEY) match
+    case null => // If no key, returns null
+      try
+        getURLContent(Globals.versionsDictionaryUrl).onComplete {
+          case Success(json: String) =>
+            window.sessionStorage.setItem(KEY, json)
+            addVersionsList(json)
+          case Failure(_) =>
+            window.sessionStorage.setItem(KEY, UNDEFINED_VERSIONS)
+            disableButton()
+        }
+      catch // Globals.versionDictionaruUrl is undefined
+        case e =>
+          window.sessionStorage.setItem(KEY, UNDEFINED_VERSIONS)
+          disableButton()
+    case value => value match
+      case UNDEFINED_VERSIONS =>
+        disableButton()
+      case json =>
+        addVersionsList(json)
+
+  document.addEventListener("click", (e: Event) => {
+    if e.target.asInstanceOf[html.Element].id != "dropdown-button" then
+      document.getElementById("dropdown-content").classList.remove("show")
+      document.getElementById("dropdown-button").classList.remove("expanded")
+  })
+
+  document.getElementById("version").asInstanceOf[html.Span].onclick = (e: Event) => {
+    e.stopPropagation
+  }
+end DropdownHandler
+
+@JSExportTopLevel("dropdownHandler")
+def dropdownHandler() =
+  if document.getElementById("dropdown-content").getElementsByTagName("a").size > 0 &&
+     window.getSelection.toString.length == 0 then
+    document.getElementById("dropdown-content").classList.toggle("show")
+    document.getElementById("dropdown-button").classList.toggle("expanded")
+
+@JSExportTopLevel("filterFunction")
```

Comment:
Where are this method and `dropdownHandler` called?

Reply:
They are called at [HtmlRenderer.scala:228](https://github.com/lampepfl/dotty/pull/12596/files#diff-7598992a2efc5327b87381e7b94f4ccd878b295ee1a7c110d8dce1dd1eed8978R228) and [HtmlRenderer.scala:232](https://github.com/lampepfl/dotty/pull/12596/files#diff-7598992a2efc5327b87381e7b94f4ccd878b295ee1a7c110d8dce1dd1eed8978R232)